File: judo-data/src/main/java/com/judopay/model/Receipts.java
package com.judopay.model;

import java.util.List;

public class Receipts {

    private final Integer resultCount;
    private final Integer pageSize;
    private final Integer offset;
    private final List<Receipt> results;

    public Receipts(Integer resultCount, Integer pageSize, Integer offset, List<Receipt> results) {
        this.resultCount = resultCount;
        this.pageSize = pageSize;
        this.offset = offset;
        this.results = results;
    }

    public Integer getResultCount() {
        return resultCount;
    }

    public Integer getPageSize() {
        return pageSize;
    }

    public Integer getOffset() {
        return offset;
    }

    public List<Receipt> getResults() {
        return results;
    }

    @Override
    public String toString() {
        return "Receipts{" +
                "results=" + results +
                ", offset=" + offset +
                ", pageSize=" + pageSize +
                ", resultCount=" + resultCount +
                '}';
    }
}

File: judokit-android/src/main/java/com/judopay/judokit/android/api/model/response/Receipts.kt
package com.judopay.judokit.android.api.model.response

/**
 * Represents the list of [Receipt] objects returned when querying the judo API for Receipts.
 */
class Receipts(
    val resultCount: Int,
    val pageSize: Int,
    val offset: Int,
    val results: List<Receipt>,
) {
    override fun toString(): String = "Receipts(resultCount=$resultCount, pageSize=$pageSize, offset=$offset, results=$results)"
}
